File: pkg/R/pda.overlay.r
pda.overlay = function(pdaList,nfine=501,ncoarse=11,...){

fdrange = pdaList$bwtlist[[1]]$fd$basis$range

tfine = seq(fdrange[1],fdrange[2],len=nfine)
beta0vals = eval.fd(tfine,pdaList$bwtlist[[1]]$fd)
beta1vals = eval.fd(tfine,pdaList$bwtlist[[2]]$fd)

plot(beta1vals,beta0vals,type='l',col=4,xlab='beta 1',ylab='beta 0',...)
abline(h=0,col=2) 
abline(v=0,col=2)

bv = seq(min(beta1vals),max(beta1vals),len=nfine)
lines(bv,(bv/2)^2,col=2,lty=2,...)

tcoarse = seq(fdrange[1],fdrange[2],len=ncoarse)
beta0valsc = eval.fd(tcoarse,pdaList$bwtlist[[1]]$fd)
beta1valsc = eval.fd(tcoarse,pdaList$bwtlist[[2]]$fd)

text(x=beta1valsc,y=beta0valsc,labels=tcoarse,col=4)

}

File: pkg/inst/Matlab/additions/pda_overlay.m
function [] = pda_overlay(bfdcell,nfine,ncoarse)

if nargin < 3, ncoarse = 11; end
if nargin < 2, nfine = 501; end

fdrange = getbasisrange(getbasis(getfd(bfdcell{1})));

tfine = linspace(fdrange(1),fdrange(2),nfine);
beta0vals = eval_fd(tfine,getfd(bfdcell{1}));
beta1vals = eval_fd(tfine,getfd(bfdcell{2}));

beta0range = [min(beta0vals),max(beta0vals)];
beta1range = [min(beta1vals),max(beta1vals)];

plot(beta1vals,beta0vals,'b','linewidth',2)
xlabel('beta 1')
ylabel('beta 0')
hold on
plot([0 0],beta0range,'r')
plot(beta1range,[0 0],'r')

bv = linspace(beta1range(1),beta1range(2),nfine);
which = (bv/2).^2 < beta0range(2);
plot(bv(which),(bv(which)/2).^2,'r')

tcoarse = linspace(fdrange(1),fdrange(2),ncoarse);
beta0valsc = eval_fd(tcoarse,getfd(bfdcell{1}));
beta1valsc = eval_fd(tcoarse,getfd(bfdcell{2}));

text(beta1valsc,beta0valsc,num2str(tcoarse'))
hold off
